### PR TITLE
Move data from ledger_state into preferences

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/9869
Resolves https://github.com/brave/brave-browser/issues/9750
Resolves https://github.com/brave/brave-browser/issues/7165

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
plan 1:
- start browser from mater
- enable rewards
- set values in AC settings
- close the browser
- switch to this PR
- start the browser
- go to AC settings and make sure that all are the same
- go rewards internals and make sure that wallet info matches as well

plan 2:
- enable rewards
- claim grant
- make sure that you see total balance with rates correctly

plan 3:
- enable rewards
- go to unverified publisher and open tip and monthly banner. Make sure that you see default values (currently 1, 10, 100) in there

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/brave_rewards/browser/rewards_service_impl.cc [hunk 1]
```diff
@@ -746,8 +744,6 @@ void RewardsServiceImpl::OnWalletInitialized(ledger::Result result) {
     ready_.Signal();
 
   if (result == ledger::Result::WALLET_CREATED) {
-    SetRewardsMainEnabled(true);
-    SetAutoContribute(true);
```
Comment (NejcZdovc) [inline components/brave_rewards/browser/rewards_service_impl.cc:750]: this fixes https://github.com/brave/brave-browser/issues/9750

File: vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.cc [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+void APIParameters::OnFetch(const ledger::UrlResponse& response) {
+  BLOG(6, ledger::UrlResponseToString(__func__, response));
+
+//  if (response.status_code != net::HTTP_OK) {
+//    SetRefreshTimer(brave_base::random::Geometric(45));
+//    return;
+//  }
+
+  const std::string& body = R"({
```
Comment (NejcZdovc) [inline vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.cc:122]: waiting for server side to be merged, so I am just using mocked one for now

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts [hunk 3]
```diff
@@ -52,13 +52,15 @@ export const rewardsPanelReducer = (state: RewardsExtension.State | undefined, a
       state.walletCorrupted = false
       break
     case types.ON_WALLET_INITIALIZED: {
+      console.log(payload)
```
Comment (emerick) [inline components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts:55]: Is this a debug remnant?

File: components/brave_rewards/browser/parameters.cc [hunk 4]
```diff
@@ -0,0 +1,21 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/parameters.cc:1]: fixed

File: components/brave_rewards/browser/parameters.cc [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/browser/parameters.h"
+
+namespace brave_rewards {
+
+Parameters::Parameters() = default;
+Parameters::~Parameters() = default;
+
+Parameters::Parameters(const Parameters &properties) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply components/brave_rewards/browser/parameters.cc:13]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.h [hunk 6]
```diff
@@ -61,7 +62,7 @@ void SetAutoContributionAmount(
     bat_ledger::LedgerImpl* ledger,
     const double amount);
 
-bool GetAutoContributionAmount(bat_ledger::LedgerImpl* ledger);
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.h]: it was moved down and renamed, still in use

File: components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts (hunk 3 above)
Comment (NejcZdovc) [reply components/brave_rewards/resources/extension/brave_rewards/background/reducers/rewards_panel_reducer.ts:55]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.cc [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+    return ledger::Result::LEDGER_ERROR;
+  }
+  parameters->auto_contribute_choice = *ac_choice;
+
+  auto* ac_choices = dictionary->FindListPath("autocontribute.choices");
+  if (!ac_choices || ac_choices->GetList().empty()) {
+    return ledger::Result::LEDGER_ERROR;
+  }
+
+  for (auto& choice : ac_choices->GetList()) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.cc:54]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.cc [hunk 8]
```diff
@@ -3,12 +3,51 @@
...
+    return {};
+  }
+
+  base::ListValue* list_value = nullptr;
+  if (!list->GetAsList(&list_value)) {
+    return {};
+  }
+
+  std::vector<double> items;
+  for (auto& item : list_value->GetList()) {
```
[the remark this answers is not in the document]
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/state/state_util.cc:42]: fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.h [hunk 9]
```diff
@@ -0,0 +1,40 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVELEDGER_API_API_PARAMETERS_H_
+#define BRAVELEDGER_API_API_PARAMETERS_H_
+
+#include "bat/ledger/ledger.h"
+#include "bat/ledger/mojom_structs.h"
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.h:10]: Isn't this already included in `ledger.h`?

File: vendor/bat-native-ledger/src/bat/ledger/internal/request/request_api.h [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVELEDGER_COMMON_REQUEST_API_H_
+#define BRAVELEDGER_COMMON_REQUEST_API_H_
+
+#include <string>
+
+#include "bat/ledger/mojom_structs.h"
```
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/request/request_api.h:11]: Is this used here?

File: vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.h (hunk 9 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/api/api_parameters.h:10]: yup, fixed

File: vendor/bat-native-ledger/src/bat/ledger/internal/request/request_api.h (hunk 10 above)
Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/request/request_api.h:11]: nope, removed

**Comment by NejcZdovc** [comment]
CI passed on window, linux and ios. Restarting macos

**Pull request merged**